www/~/%username/routes/associate.json.spt

```diff
@@ -40,7 +40,13 @@ if change:
     elif network == 'paypal':
         if address not in participant.get_verified_email_addresses():
             raise Response(400, _("Only verified email addresses allowed."))
-        ExchangeRoute.insert(participant, network, address)
+        try:
+            ExchangeRoute.insert(participant, network, address)
+        except IntegrityError as e:
+            if e.pgcode != errorcodes.UNIQUE_VIOLATION:
+                raise
+            existing = ExchangeRoute.from_address(participant, network, address)
```

Comment:
if I'm not mistaken, this occurs when insertion fails. Could we check before insertion instead? And one nit: rename existing to existing_route?